docs/design-proposals/multi-tenancy/multi-tenancy-api-resource-model.md

```diff
@@ -159,14 +159,88 @@ Here are some examples:
+
+ 3. **Autonomy**. The tenant admin has the final say on what CRD will be used in his tenant space when there is an overlap between the tenant's CRD and that of the system tenant. The tenant can choose MultiTenancyCRD policy to be :
+  * NeverUseSystemCRD
+  * SystemCRDFirst
+  * LocalCRDFirst
+
+  Besides, the tenant user can change the MultiTenancyCRD policy anytime. 
+
+  The default policy will be LocalCRDFirst. Under the default policy, if a tenant has deployed CRD objects and CRD operators based on its own CRD, the introduction of an overlapping system CRD will not cause break.
+
+ 4. **Backward compatibility**. The legacy CRD definitions and CRD operators continue to work in Arktos. More specifically, it means:
+    * An existing defintion of CRD (usually in a .yaml file) can be applied without any change.
+    * An existing CRD operator image can work in Arktos without re-compiling or image rebuilding.
+    * An existing CRD operator source code build and work in Arktos. So developers can create new operators or revise existing operators in Arktos without learning new APIs.
+
+However, we need to point out that, Arktos does not allow a regular tenant to access cluster-scope resources, such as nodes and daemonsets. So a CRD definition or a CRD operator that needs access to the cluster-scope resources will no longer be supported in a regular tenant's space. Surely, they can continue to work in the system tenant's space. For the same reason, a regular tenant is not allowed to create a cluster-scope CRD. If regular tenant specifies its CRD to be cluster-scoped, the CRD will be changed to tenant-scoped. 
  
- (**TBD: Requires more implementation details**)
+ 
+#### Detailed Design
+
+ The key changes to realize the isolation principles are:
+ * CustomResourceDefinition will be changed to a tenant-scoped resource. As a result, same CRD G/V/K can co-exist as long as they live under different tenants. 
+ * APIResource struct ( which is the data structure to holds the info of resource types suspported in API server) will have a new field, Tenant. For non-CRD resources, this field is empty, which means the resources are applicable to all tenants. For CRD resources, it indicates the belonging tenant.
+ * The CRD shown in the response to a resource discovery request should only show those usable to tenant in the request.
 
+The key changes to realize managability and automony are:
+* A new field, ApplicableRange, will be added to the CustomResourceDefinitionSpec. The field is ignored in a regular-tenant CRD. Yet for a system-tenant CRD, it indicates the scope of the regular tenants that this CRD is applicable to. This field is a list of annotations, which are used to finding matching tenants. (other designs may also work, as long as the field give info in finding matching tenants)
```

Comment:
This only solves the problem of discovering CR resources. But it requires code changes in operator to handle multi-tenancy CRs.

Reply:
Yes, the existing operator only works in a back-compatible way within his own tenant'world. If an operator needs access multi-tenant CRs, changes are inevitable. 

I added a few lines in "backward-compatibility" to highlight this.